### ISSUE TITLE
Building from source on OSX with GNU compiler

### ISSUE THREAD

**Issue by nishbo** [simbody/simbody#350]
For some reason, when building simbody with a gcc compiler instead of a native one on OSX, unrecognized option `-stdlib=libc++` produces an error instead of a warning. This is a quick fix: replace line 270 in `CMakeList.txt` with:
```
            IF(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
                SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
            ELSE()
                SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
                SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
            ENDIF()
```

**Comment by scpeters** [comment]
Is this equivalent to what you mean?
~~~
diff --git a/CMakeLists.txt b/CMakeLists.txt
index f41a9a4..6c01822 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -268,7 +268,9 @@ IF(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU" OR
             SET(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY "libc++")
         ELSE()
             SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-            SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+            IF(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
+              SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+            ENDIF()
         ENDIF()
     ELSE() # not APPLE
         SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
~~~

**Comment by nishbo** [comment]
@scpeters Yes, it is. If `-stdlib=libc++` style is only used by Clang, than it is a solution.

**Comment by scpeters** [comment]
`libc++` is used by clang (maybe others?) but gcc uses `libstdc++`

**Comment by nishbo** [comment]
Typo?: gcc uses `-std=c++11`, not `-stdlib=libc++` on every system, I believe.

**Comment by scpeters** [comment]
I was referring to its name, not the compiler argument: https://gcc.gnu.org/libstdc++/

**Comment by chrisdembia** [comment]
Great catch. @nishbo would you be interested in submitting a PR for this?

**Comment by nishbo** [comment]
Thanks @chrisdembia, hope I made correct PR.

**Comment by nishbo** [comment]
Issue number in PR name did not get referenced.
https://github.com/simbody/simbody/pull/352

**Comment by sherm1** [comment]
I'm assuming this is fixed now, by PR #352. @nishbo, please re-open if not.
